src/core_ghidra.cpp

```diff
@@ -132,109 +133,120 @@ static void ApplyPrintCConfig(RConfig *cfg, PrintC *print_c)
 	print_c->setMaxLineSize(cfg_var_linelen.GetInt(cfg));
 }
 
-static void Decompile(RCore *core, DecompileMode mode)
-{
-	DecompilerLock lock;
-
-#ifndef DEBUG_EXCEPTIONS
+static void refactored_decompile(RCore *&core, std::stringstream &out_stream,
```

Comment:
Better signature:
```C++
static void Decompile(RCore *core, ut64 addr, DecompileMode mode, std::stringstream &out_stream, RAnnotatedCode **out_code)
```
The `&` for core is unnecessary, for code it's better readable like this (you will have to use `*out_code` instead of `code` in the function body).
Also this makes is more clear what the inputs and what the outputs are.

Reply:
Changed.